static/js/message_live_update.js

```diff
@@ -28,5 +28,11 @@ export function update_avatar(user_id, avatar_url) {
     let url = avatar_url;
     url = people.format_small_avatar_url(url);
     message_store.update_property("small_avatar_url", url, {user_id});
+
+    rerender_messages_view();
+}
+
+export function update_user_status_emoji(user_id, status_emoji_info) {
+    message_store.update_property("status_emoji_info", status_emoji_info, {user_id});
     rerender_messages_view();
 }
```

Comment:
I think this implementation, as written, will trigger a full rerender of the displayed message list every time someone changes their status, regardless of whether any messages sent by that user were visible.

This was OK previously because fields like name very rarely change.  But we probably will want to make the rerender conditional on the user whose status was changed actually having any messages that they'd sent.

Let's split fixing this out as a small preparatory PR that changes this interface here with the existing "change name" code path -- I think basically we want to replace `rerender_messages_view` function in this module to instead be `rerender_messages_view_if_has_sender(user_id)`, and then to add a conditional that calls into a `message_list_data.js` function.  You can probably use this as inspiration:
```
    get_last_message_sent_by_me() {                                                                      
        const msg_index = _.findLastIndex(this._items, {sender_id: page_params.user_id});                
        if (msg_index === -1) {                                                                          
            return undefined;                                                                            
        }                                                                                                
        const msg = this._items[msg_index];                                                              
        return msg;                                                                                      
    }                                                                                                    
```

Reply:
(working on it)

Reply:
opened #21081